themes/classic/templates/catalog/_partials/product-prices.tpl

```diff
@@ -50,7 +50,7 @@
 
     {block name='product_unit_price'}
       {if $displayUnitPrice}
-        <p class="product-unit-price">{convertPrice price=$unit_price} {l s='per %s' sprintf=$product.unity}</p>
+        <p class="product-unit-price"{hook h='displayProductPriceBlock' product=$product type="unit_price" returntype="style"}>{convertPrice price=$unit_price} {l s='per %s' sprintf=$product.unity}</p>
```

Comment:
Ok, there is another solution! But not right now.

Please just remove this hook on this pull request and create another hook useful for everyone like this:

```smarty
{block name='product_unit_price'}
  {if $displayUnitPrice}
    <p class="product-unit-price">{convertPrice price=$unit_price} {l s='per %s' sprintf=$product.unity}</p>
    {hook h='displayProductPriceBlock' product=$product type="unit_price"}
  {else}
    {hook h='displayProductPriceBlock' product=$product type="unit_price_not_displayed"}
  {/if}
{/block}
```

Then, we will add a hook on the product presenter so people should modify for interesting values like, for you, `displayUnitPrice` (currently on the controller but we should move it).

What do you think? Way better IMO.

I know we have to be fast for you, it will be done today.